src/labone/nodetree/node.py

```diff
@@ -0,0 +1,863 @@
+        """Type of the node."""
+        return self._info["Type"]
+
+    @property
+    def unit(self) -> str:
+        """Unit of the node."""
+        return self._info["Unit"]
+
+    _option_info = namedtuple("_option_info", ["enum", "description"])
+
+    @cached_property
+    def options(self) -> t.Dict[int, _option_info]:
+        """Options of the node."""
+        option_map = {}
+        for key, value in self._info.get("Options", {}).items():
+            # Find all the keywords. We use only the first one
+            # since it should be unambiguous
+            enum_re = re.findall(r'"([a-zA-Z0-9-_"]+)"', value)
+            enum = enum_re[0] if enum_re else ""
+
+            # The description is either what comes after
+            # the colon and space, or the whole string.
+            # This is the case for nameless options, when the
+            # key is an integer (for example demods/x/order)
+            desc = re.findall(r'(?:.+":\s)?(.+)$', value)[0]
+
+            option_map[int(key)] = self._option_info(enum, desc)
+        return option_map
+
+    @cached_property
+    def enum(self) -> t.Optional[NodeEnum]:
+        """Enum of the node options."""
+        options_reversed = {}
+        for int_key, value in self._info.get("Options", {}).items():
+            # Find all the keywords associated to a integer key
+            enum_re = re.finditer(r'"(?P<keyword>[a-zA-Z0-9-_"]+)"', value)
+            for m in enum_re:
+                keyword = m.group("keyword")
+                options_reversed[keyword] = int_key
+
+        return (
+            NodeEnum(self.path, options_reversed, module=__name__)
+            if options_reversed
+            # Nameless options do not have a enum.
+            else None
+        )
+
+
+class MetaNode(ABC):
+    """Basic functionality for tree of nodes.
+
+    This class provides common behaviour for the Node and ResultNode classes.
+    """
+
+    def __init__(self,
+                 path_segments: t.Tuple[t.Union[str, int]],
+                 subtree_structure: NestedDict,
+                 path_aliases: dict[tuple[NormalizedPathSegment], tuple[NormalizedPathSegment]],
+                 guarantee_valid_path_format: bool = False
+                 ):
+
+        if not guarantee_valid_path_format:  # todo is this option worth it
+            path_segments = tuple(normalize_path_segment(e) for e in path_segments)
+
+        self._path_segments: tuple[NormalizedPathSegment] = path_segments
+        self._path_aliases = path_aliases
+
+        # redirecting path_segments aliases to original path_segments
+        while self._path_segments in self._path_aliases.keys():
+            self._path_segments = self._path_aliases[self._path_segments]
+
+        self._subtree_structure = subtree_structure
+        self._contains_wildcards = any(
+            [segment == WILDCARD for segment in self.path_segments])  # todo optimize via argument?
+        self._is_leaf = self._subtree_structure == {}
```

Comment:
As discussed we want to try out if we can create separate classes for leaf nodes and non leaf nodes